### ISSUE TITLE
[postgresql] Support the `USING <method>` option for `CREATE TABLE`

### ISSUE THREAD

**Issue by edgarrmondragon** [sqlalchemy/sqlalchemy#10904]
### Describe the use case

The Postgre `CREATE TABLE` supports a `USING ...` option:

> This optional clause specifies the table access method to use to store the contents for the new table; the method needs be an access method of type TABLE. See [Chapter 63](https://www.postgresql.org/docs/current/tableam.html) for more information. If this option is not specified, the default table access method is chosen for the new table. See [default_table_access_method](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-DEFAULT-TABLE-ACCESS-METHOD) for more information.

Some variants of Postgres use this to allow creating tables with different access methods, e.g. `columnar`.

https://www.postgresql.org/docs/current/sql-createtable.html#SQL-CREATETABLE-METHOD

### Databases / Backends / Drivers targeted

PostgreSQL

### Example Use

```python
Table("my_columnar_table", metadata, ..., postgresql_using="columnar")
```

Which would translate to

```sql
CREATE TABLE my_columnar_table (...) USING columnar;
```

### Additional context

The overall purpose is to support [Hydra databases](https://docs.hydra.so/concepts/using-hydra-columnar), so I may need to create a new dialect anyway but this still would be nice to have for the builtin postgres dialect.

Is this as easy as updating https://github.com/sqlalchemy/sqlalchemy/blob/b455b1fc03e25fc6a53cac74f3cee515c3efbf8a/lib/sqlalchemy/dialects/postgresql/base.py#L2381

and

https://github.com/sqlalchemy/sqlalchemy/blob/b455b1fc03e25fc6a53cac74f3cee515c3efbf8a/lib/sqlalchemy/dialects/postgresql/base.py#L3001-L3010

?

**Comment by CaselIT** [comment]
Hi,

Yes that would be the way to do it.

@zzzeek there doesn't seem to be an open ended "siffixes" in the table definition, that could be be used for cases like these. Do you think it would make sense? A bit similar to the "prefixes" kw

**Comment by zzzeek** [comment]
prefixes is a hack that doesnt work with things like cross-database support or reflection so if we can just add support for kws that would otherwise be suffixes that would be best.  also "suffixes" would have to interact with other fixed directives and it's messy to figure out how that would work anyway.

i think there's already an issue for PG USING for tables that would make this a dupe.

**Comment by CaselIT** [comment]
> prefixes is a hack that doesnt work with things like cross-database support or reflection so if we can just add support for kws that would otherwise be suffixes that would be best. also "suffixes" would have to interact with other fixed directives and it's messy to figure out how that would work anyway.

yes prefixes is kinda strange since it takes dialect specific stuff without a way of specify the dialect. Let's add this then

> i think there's already an issue for PG USING for tables that would make this a dupe.

I could not find one. Maybe you remembered this one that talks about using but it's in something else? https://github.com/sqlalchemy/sqlalchemy/issues/8130

**Comment by zzzeek** [comment]
yah could not find the dupe either so this is it.

**Comment by sqla-tester** [comment]
**Edgar Ramírez Mondragón** has proposed a fix for this issue in the **main** branch:

**Support specifying access method when creating Postgres tables** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5114

**Comment by sqla-tester** [comment]
**Edgar Ramírez Mondragón** has proposed a fix for this issue in the **rel_2_0** branch:

**Support specifying access method when creating Postgres tables** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5115
